### PR TITLE
Add excution error result

### PR DESCRIPTION
**Pull request by bjk7119**
## Description

- Add excution error result
- Bug fix about output file name when using -o option

< Example of xml >
```
<?xml version='1.0' encoding='UTF-8'?>
<results><error id="rule_key_osc_checker_01" line="0" msg="&#10;# SUMMARY&#10;* Open Source Package File: &#10;                                * Detected Licenses : -, GPL-3.0-only&#10;                                * Files without copyright : 5 / 16&#10;                                * Files without license : 5 / 16&#10;&#10;&#10;Ref. Copyright and License Writing Rules in Source Code. : https://oss.lge.com/guide/process/osc_process/1-identification/copyright_license_rule.html" /><execution_errors><execution_error>Can't find oss-pkg-info file</execution_error><execution_error>Can't find a path</execution_error></execution_errors></results>
```

< Example of yaml >
```
Checking copyright/license writing rules:
  Compliant: Not OK
  Summary:
    Open Source Package File: []
    Detected Licenses:
    - '-'
    - GPL-3.0-only
    Count without license / Total: 5 / 16
    Count without copyright / Total: 5 / 16
  Files without license and copyright:
  - fosslight_reuse.egg-info/SOURCES.txt
  - fosslight_reuse.egg-info/dependency_links.txt
  - fosslight_reuse.egg-info/entry_points.txt
  - fosslight_reuse.egg-info/requires.txt
  - fosslight_reuse.egg-info/top_level.txt
  Files without license:
  - fosslight_reuse.egg-info/SOURCES.txt
  - fosslight_reuse.egg-info/dependency_links.txt
  - fosslight_reuse.egg-info/entry_points.txt
  - fosslight_reuse.egg-info/requires.txt
  - fosslight_reuse.egg-info/top_level.txt
  Files without copyright:
  - fosslight_reuse.egg-info/SOURCES.txt
  - fosslight_reuse.egg-info/dependency_links.txt
  - fosslight_reuse.egg-info/entry_points.txt
  - fosslight_reuse.egg-info/requires.txt
  - fosslight_reuse.egg-info/top_level.txt
  Tool Info:
    OS Info: Linux 4.15.0-144-generic
    Path to analyze: /home/jaekwonbang/fosslight_reuse_0512/src
    Python Version: 3
    FL Reuse Version: fosslight_reuse v2.1.9
  Excution Error:
  - Can't make result file
  - Can't find oss pakage file
```

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [x] Refactoring, Maintenance
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/fosslight_reuse/_result.py [hunk 1]
```diff
@@ -122,14 +124,19 @@ def result_for_xml(result_item: ResultItem):
                 items.set('line', '0')
                 items.set('msg', MSG_FOLLOW_LIC_TXT)
                 _root_xml_item.append(items)
+
+    if result_item._excution_error != "N/A":
```
Comment (soimkim) [inline src/fosslight_reuse/_result.py:128]: if result_item._excution_error != "N/A": -> if result_item._excution_error:

File: src/fosslight_reuse/_result.py [hunk 2]
```diff
@@ -122,14 +124,18 @@ def result_for_xml(result_item: ResultItem):
                 items.set('line', '0')
                 items.set('msg', MSG_FOLLOW_LIC_TXT)
                 _root_xml_item.append(items)
+
+    if result_item._excution_error:
+        error_xml = ET.Element('execution_error')
+        error_xml.text = result_item._excution_error
```
Comment (soimkim) [inline src/fosslight_reuse/_result.py:130]: Isn't _execution_error a list?

**Review by soimkim — CHANGES_REQUESTED** [review]

**Comment by soimkim** [comment]
@bjk7119 , Please add examples with more than two execution errors to the description in both yaml and xml.

File: src/fosslight_reuse/_result.py (hunk 2 above)
Comment (bjk7119) [reply src/fosslight_reuse/_result.py]: yes, execution_error is list type

**Review by soimkim — APPROVED** [review]

**Comment by soimkim** [comment]
@bjk7119 , Please update xml format in the description.

**Pull request merged**